### PR TITLE
Feature/add notifier discord

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

implement #3181

### CODE REVIEW

File: medusa/__main__.py [hunk 1]
```diff
@@ -1755,6 +1764,14 @@ def save_config():
         new_config['Telegram']['telegram_id'] = app.TELEGRAM_ID
         new_config['Telegram']['telegram_apikey'] = app.TELEGRAM_APIKEY
 
+        new_config['Discord'] = {}
+        new_config['Discord']['use_discord'] = int(app.USE_DISCORD)
+        new_config['Discord']['discord_notify_onsnatch'] = int(app.DISCORD_NOTIFY_ONSNATCH)
+        new_config['Discord']['discord_notify_ondownload'] = int(app.DISCORD_NOTIFY_ONDOWNLOAD)
+        new_config['Discord']['discord_notify_onsubtitledownload'] = int(app.DISCORD_NOTIFY_ONSUBTITLEDOWNLOAD)
+        new_config['Discord']['discord_webhook'] = app.DISCORD_WEBHOOK
+        new_config['Discord']['discord_tts'] = app.DISCORD_TTS
```
Comment (medariox) [inline medusa/__main__.py]: `app.DISCORD_TTS` is a boolean here. Don't we need an int, like the rest (`int(app.DISCORD_TTS)`)?

File: medusa/app.py [hunk 2]
```diff
@@ -392,6 +392,15 @@ def __init__(self):
         self.TELEGRAM_ID = ''
         self.TELEGRAM_APIKEY = ''
 
+        self.USE_DISCORD = False
+        self.DISCORD_NOTIFY_ONSNATCH = False
+        self.DISCORD_NOTIFY_ONDOWNLOAD = False
+        self.DISCORD_NOTIFY_ONSUBTITLEDOWNLOAD = False
+        self.DISCORD_WEBHOOK = None
+        self.DISCORD_NAME = 'pymedusa'
+        self.DISCORD_AVATAR_URL = 'https://raw.githubusercontent.com/pymedusa/medusa/master/themes-default/slim/static/images/ico/favicon-144.png'
```
Comment (medariox) [inline medusa/app.py]:
```suggestion
        self.DISCORD_AVATAR_URL = '{base_url}/images/ico/favicon-144.png'.format(base_url=self.BASE_PYMEDUSA_URL)
```

Comment (p0psicles) [reply medusa/app.py]: Hmm. `https://raw.githubusercontent.com` is something different then our cdn. Don't really feel much for creating a new constant specially for this?

**Comment by p0psicles** [comment]
One thing medariox, I didn't to any unicode conversions. Should I?

File: medusa/app.py (hunk 2 above)
Comment (medariox) [reply medusa/app.py]: But it points to the same image?

**Comment by medariox** [comment]
@p0psicles 
Which conversions do you mean? I believe we have to explicitly encode the payload: https://github.com/pymedusa/Medusa/pull/7189/files#diff-249cf00d117c6ee01fcc6a365e5172c6R46-R51
Otherwise requests will attempt an implicit latin-1 encode.

File: medusa/app.py (hunk 2 above)
Comment (p0psicles) [reply medusa/app.py]: I tried it. But didn't work

Comment (medariox) [reply medusa/app.py]: Strange.
https://cdn.pymedusa.com/images/ico/favicon-144.png
and
https://raw.githubusercontent.com/pymedusa/medusa/master/themes-default/slim/static/images/ico/favicon-144.png

both work for me.

**Comment by p0psicles** [comment]
@medariox encode the entire payload or just the message?

**Comment by p0psicles** [comment]
This PR should be finished now

**Pull request merged**